group-generators/helpers/data-providers/tally/provider.ts

```diff
@@ -0,0 +1,269 @@
+    return this.query(
+      gql`
+        query Governors {
+          governors(chainIds: "eip155:1") {
+            id
+            name
+          }
+        }
+      `,
+      {}
+    );
+  }
+
+  public async getProposalVoters({
+    name,
+    proposalId,
+  }: inputGetProposalVoters): Promise<FetchedData> {
+    const data = await this._queryNameToGovernorAddress();
+
+    const fetchedData: { [address: string]: number } = {};
+
+    if (data && Array.isArray(data.governors)) {
+      const matchingGovernors = data.governors.filter(
+        (governor: { name: string }) => governor.name === name
+      );
+
+      for (const governor of matchingGovernors) {
+        for (const proposals of governor.proposals) {
+          const propId = proposals.id;
+
+          if (propId == proposalId) {
+            const allVotes = proposals.votes;
+
+            allVotes.forEach((voteItem: { id: string; voter: any }) => {
+              const { voter } = voteItem;
+              const { address } = voter;
+              if (fetchedData[address]) {
+                fetchedData[address]++;
+              } else {
+                fetchedData[address] = 1;
+              }
+            });
+          }
+        }
+      }
+
+      return fetchedData;
+    }
+
+    return fetchedData;
+  }
+
+  public async getProposalVotersCount({
+    name,
+    proposalId,
+  }: inputGetProposalVoters): Promise<number> {
+    const ProposalVoterData = await this.getProposalVoters({ name, proposalId });
+
+    const ProposalVoterCount = Object.keys(ProposalVoterData).length;
+    return ProposalVoterCount;
+  }
+
+  public async getGovernanceVoters({ name }: inputGetGovernanceProposers): Promise<FetchedData> {
+    const data: queryData = await this._queryNameToGovernorAddress();
+    const fetchedData: { [address: string]: number } = {};
+
+    if (data && Array.isArray(data.governors)) {
+      const matchingGovernors = data.governors.filter(
+        (governor: { name: string }) => governor.name === name
+      );
+
+      for (const governor of matchingGovernors) {
+        for (const proposals of governor.proposals) {
+          const votesArray = proposals?.votes;
+
+          votesArray.forEach((item: { id: string; voter: any }) => {
+            const { voter } = item;
+            const { address } = voter;
+            if (fetchedData[address]) {
+              fetchedData[address]++;
+            } else {
+              fetchedData[address] = 1;
+            }
+          });
+        }
+      }
+
+      return fetchedData;
+    }
+
+    return fetchedData;
+  }
+
+  public async getGovernanceVotersCount({ name }: inputGetGovernanceProposers): Promise<number> {
+    const VoterData = await this.getGovernanceVoters({ name });
+
+    const VoterCount = Object.keys(VoterData).length;
+    return VoterCount;
+  }
+
+  public async getGovernanceDelegates({ name }: inputGetGovernanceProposers): Promise<FetchedData> {
```

Comment:
This function takes too much time when processed. If the limit is stuck at 30, we could remove this function from the scope of the PR.